### DISCUSSION TITLE
How do I get the following info?

### DISCUSSION THREAD

**Discussion by ghost** [dahall/TaskScheduler#947] [Q&A] [answered]
Hello,

I'm trying to use this library to get some info from a remote server's Task Scheduler.

I'm having trouble getting time and frequency of job; list of actions with their details.

Any help appreciated.

**Comment by ghost** [comment]
Well, I figured out the first one, but I don't see how I can get list of actions.

I tried foreaching the Definition.Actions property, but it only contains ActionType and Id.

I need list of actions and their details (e.g., Start a program -> C:\Progs\program.cmd")

**Comment by dahall** [comment, marked as answer]
You need to cast the `Action` to the type corresponding to the value of `ActionType`.

**Reply by ghost** [reply]
Could you please give an example?

**Reply by ghost** [reply]
I think I figured it out.

![image](https://user-images.githubusercontent.com/5992169/173512077-9f3b5eda-1439-4421-812c-a8d17beeb8f6.png)

Is this what you meant?

**Reply by dahall** [reply]
Precisely

**Reply by ghost** [reply]
Thank you very much, I had trouble understanding what you meant :)

I got the info I needed.
